CMakeLists.txt

```diff
@@ -222,32 +229,69 @@ if(NOT NETWORKIT_EXT_TLX)
 	endif()
 else()
 	add_library(tlx STATIC IMPORTED)
-	set_target_properties(tlx PROPERTIES
-			IMPORTED_LOCATION "${NETWORKIT_EXT_TLX}/lib/libtlx.a"
-			INTERFACE_INCLUDE_DIRECTORIES "${NETWORKIT_EXT_TLX}/include/")
+	if(WIN32)
+		set_target_properties(tlx PROPERTIES
+				IMPORTED_LOCATION "${NETWORKIT_EXT_TLX}/lib/tlx.lib"
+				INTERFACE_INCLUDE_DIRECTORIES "${NETWORKIT_EXT_TLX}/include/")
+	else()
+		set_target_properties(tlx PROPERTIES
+				IMPORTED_LOCATION "${NETWORKIT_EXT_TLX}/lib/libtlx.a"
+				INTERFACE_INCLUDE_DIRECTORIES "${NETWORKIT_EXT_TLX}/include/")
+	endif()
 endif()
 
 ################################################################################
 # NETWORKIT MODULES
 if(NETWORKIT_BUILD_CORE AND NETWORKIT_MONOLITH)
-	if(NETWORKIT_STATIC)
-		add_library(networkit networkit/cpp/networkit.cpp)
+
+	if(WIN32)
+		# Static and non-static is ignored for Windows-builds. Shared-build only is not supported and getting linked against multiple dll's 
+		# needs a dll-based globalstate to enable consistent global variable share.
+		add_library(networkit_export SHARED networkit/cpp/GlobalState.cpp)
```

Comment:
Instead of calling the library `networkit_export`, I would prefer `networkit_state`.